# Homework Assignment - Lesson 1: Introduction to R

**Student Name:** [Nick Eisenheim]  
**Date:** [September 4 2025]  
**Course:** Data Management  

---

## Instructions

Complete all the tasks below by adding your R code in the code cells and your written responses in markdown cells. Make sure to run all cells to show your output.

**📋 SUBMISSION**: When you're done, see [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for complete submission instructions.

---

## Part 1: Setting Up Your Environment and Importing Data

### 1.1 Working Directory

Check your current working directory using `getwd()`. You should be in the main repository folder.

In [1]:
# Check current working directory
getwd()

[1] "/workspaces/assignment-1-nheisenheim/assignment/Homework"

### 1.2 Package Loading

Load the required packages: `tidyverse` and `readxl`

In [2]:
# Load required packages
library(tidyverse)
library(readxl)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.1.0     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


### 1.3 Data Import - CSV

Import the `sales_data.csv` file from the `data/` folder into a data frame named `sales_df`.

In [5]:
# Import sales data
sales_df <- read_csv("/workspaces/assignment-1-nheisenheim/data/sales_data.csv")
print("Sales data imported successfully!")

Rows: 100 Columns: 6


── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): Product, Region
dbl  (3): TransactionID, Amount, Quantity
date (1): Date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] "Sales data imported successfully!"


### 1.4 Data Import - Excel

Import both sheets from the `customer_feedback.xlsx` file.

In [7]:
# Import ratings data from Excel
ratings_df <- read_excel("/workspaces/assignment-1-nheisenheim/data/customer_feedback.xlsx", sheet = "ratings")

# Import comments data from Excel  
comments_df <- read_excel("/workspaces/assignment-1-nheisenheim/data/customer_feedback.xlsx", sheet = "customer_feedback")

print("Excel data imported successfully!")

[1] "Excel data imported successfully!"


---

## Part 2: Basic Data Inspection

For each of the three data frames you imported, perform the following inspection tasks. Add both your R code and a brief interpretation of what you observe.

### 2.1 Sales Data Inspection

In [ ]:
# First 10 rows of sales data
print("First 10 rows of sales_df:")
head(sales_df, 10)


[1] "First 10 rows of sales_df:"


TransactionID Date       Product  Amount  Quantity Region
1   1            2023-01-27 Keyboard 1039.05 1        South 
2   2            2023-05-18 Tablet    597.55 5        East  
3   3            2023-02-14 Keyboard 1539.48 1        West  
4   4            2023-07-28 Mouse     261.66 1        South 
5   5            2023-05-17 Monitor  1047.73 1        South 
6   6            2023-02-25 Keyboard 1824.05 2        East  
7   7            2023-01-20 Monitor   475.83 1        North 
8   8            2023-07-14 Mouse     758.05 5        North 
9   9            2023-10-02 Mouse    1717.20 4        South 
10 10            2023-10-15 Laptop   1437.21 3        North

In [9]:
# Structure of sales data
print("Structure of sales_df:")
str(sales_df)

[1] "Structure of sales_df:"
spc_tbl_ [100 × 6] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ TransactionID: num [1:100] 1 2 3 4 5 6 7 8 9 10 ...
 $ Date         : Date[1:100], format: "2023-01-27" "2023-05-18" ...
 $ Product      : chr [1:100] "Keyboard" "Tablet" "Keyboard" "Mouse" ...
 $ Amount       : num [1:100] 1039 598 1539 262 1048 ...
 $ Quantity     : num [1:100] 1 5 1 1 1 2 1 5 4 3 ...
 $ Region       : chr [1:100] "South" "East" "West" "South" ...
 - attr(*, "spec")=
  .. cols(
  ..   TransactionID = col_double(),
  ..   Date = col_date(format = ""),
  ..   Product = col_character(),
  ..   Amount = col_double(),
  ..   Quantity = col_double(),
  ..   Region = col_character()
  .. )
 - attr(*, "problems")=<externalptr> 


In [10]:
# Summary statistics for sales data
print("Summary statistics for sales_df:")
summary(sales_df)

[1] "Summary statistics for sales_df:"


 TransactionID         Date              Product              Amount       
 Min.   :  1.00   Min.   :2023-01-01   Length:100         Min.   :  58.06  
 1st Qu.: 25.75   1st Qu.:2023-03-20   Class :character   1st Qu.: 520.79  
 Median : 50.50   Median :2023-07-03   Mode  :character   Median : 960.75  
 Mean   : 50.50   Mean   :2023-06-27                      Mean   : 962.71  
 3rd Qu.: 75.25   3rd Qu.:2023-09-25                      3rd Qu.:1384.63  
 Max.   :100.00   Max.   :2023-12-26                      Max.   :1985.60  
    Quantity       Region         
 Min.   :1.00   Length:100        
 1st Qu.:1.00   Class :character  
 Median :3.00   Mode  :character  
 Mean   :2.69                     
 3rd Qu.:4.00                     
 Max.   :5.00                     

**Your observations about sales_df:**
# The average quantity of product sold was 2.69. THe region that sould the most product was the South and the region that sold the least was the West. The total average amount of product sold was 962.71 with the highest being 1985.60.

[Write your observations here - what do you notice about the data types, number of rows/columns, any missing values, etc.]

### 2.2 Ratings Data Inspection

In [15]:
# TODO: Add your code here for ratings_df inspection
# - First 10 rows
# - Structure  
# - Summary statistics
print("First 10 rows ratings_df")
head(ratings_df,10)
str(ratings_df)


print("Summary Statistics ratings_df")
summary(ratings_df)

[1] "First 10 rows ratings_df"


CustomerID ProductRating ServiceRating OverallSatisfaction
1   1         1             4             3                  
2   2         3             2             4                  
3   3         1             5             5                  
4   4         1             2             3                  
5   5         5             1             2                  
6   6         4             2             2                  
7   7         3             5             5                  
8   8         5             2             5                  
9   9         4             1             3                  
10 10         4             1             2                  
   RecommendationScore
1   2                 
2   7                 
3   1                 
4   6                 
5   2                 
6   6                 
7   5                 
8  10                 
9   6                 
10  0

tibble [50 × 5] (S3: tbl_df/tbl/data.frame)
 $ CustomerID         : num [1:50] 1 2 3 4 5 6 7 8 9 10 ...
 $ ProductRating      : num [1:50] 1 3 1 1 5 4 3 5 4 4 ...
 $ ServiceRating      : num [1:50] 4 2 5 2 1 2 5 2 1 1 ...
 $ OverallSatisfaction: num [1:50] 3 4 5 3 2 2 5 5 3 2 ...
 $ RecommendationScore: num [1:50] 2 7 1 6 2 6 5 10 6 0 ...
[1] "Summary Statistics ratings_df"


   CustomerID    ProductRating ServiceRating  OverallSatisfaction
 Min.   : 1.00   Min.   :1.0   Min.   :1.00   Min.   :1.00       
 1st Qu.:13.25   1st Qu.:2.0   1st Qu.:1.00   1st Qu.:2.00       
 Median :25.50   Median :3.0   Median :3.00   Median :3.00       
 Mean   :25.50   Mean   :3.2   Mean   :2.94   Mean   :2.92       
 3rd Qu.:37.75   3rd Qu.:5.0   3rd Qu.:5.00   3rd Qu.:4.00       
 Max.   :50.00   Max.   :5.0   Max.   :5.00   Max.   :5.00       
 RecommendationScore
 Min.   : 0.0       
 1st Qu.: 1.0       
 Median : 4.5       
 Mean   : 4.2       
 3rd Qu.: 7.0       
 Max.   :10.0       

**Your observations about ratings_df:**

# The average product rating is 3.2 which is higher than the average for the service which is a 2.94. Both of those are above the median but the recommendation score is a 4.2 which is below the median. 

### 2.3 Comments Data Inspection

In [16]:
# TODO: Add your code here for comments_df inspection
# - First 10 rows
# - Structure
# - Summary statistics
print("First 10 rows comments_df")
head(comments_df,10)
str(comments_df)

[1] "First 10 rows comments_df"


FeedbackID CustomerID Feedback_Text                   
1   1         12         Highly recommend this item      
2   2         40         Excellent service               
3   3         34         Poor quality control            
4   4          1         average product, nothing special
5   5         47         AMAZING customer support!!!     
6   6         13         AMAZING customer support!!!     
7   7         13         average product, nothing special
8   8         37         good VALUE for money            
9   9         49         Highly recommend this item      
10 10         23         Highly recommend this item      
   Contact_Info           Feedback_Date
1  bob.wilson@test.org    2024-02-23   
2  555-123-4567           2024-01-21   
3  jane_smith@company.com 2023-09-02   
4  jane_smith@company.com 2023-08-21   
5  555-123-4567           2023-04-24   
6  john.doe@email.com     2023-04-16   
7  (555) 987-6543         2024-03-13   
8  (555) 987-6543         2023-07-15   
9  invalid-email          2023-03-03   
10 john.doe@email.com     2023-10-18

tibble [100 × 5] (S3: tbl_df/tbl/data.frame)
 $ FeedbackID   : num [1:100] 1 2 3 4 5 6 7 8 9 10 ...
 $ CustomerID   : num [1:100] 12 40 34 1 47 13 13 37 49 23 ...
 $ Feedback_Text: chr [1:100] "Highly recommend this item" "Excellent service" "Poor quality control" "average product, nothing special" ...
 $ Contact_Info : chr [1:100] "bob.wilson@test.org" "555-123-4567" "jane_smith@company.com" "jane_smith@company.com" ...
 $ Feedback_Date: POSIXct[1:100], format: "2024-02-23" "2024-01-21" ...


**Your observations about comments_df:**

[Write your observations here]

**Reflection Question 1:** What did you observe about the structure and content of the data files you loaded? What insights can you share about the quality and organization of the data?

**Your Answer:**

[Write your response here]

### Question 1: Data Types Analysis

Based on your inspection of `sales_df`, what are the data types of the `Date` and `Amount` columns? Are these data types appropriate for typical business analytics tasks involving sales data? Explain why or why not.

**Your Answer:**

# I believe that the 'Date' column is important because it will allow you to look at past and future data to compare how your sales have changed. With that data you can establish if there are going to be trends or seasonal changes and can order inventory accordingly. The 'Amount' column is not as necessary and needs existing business knowledge to understand. 

### Question 2: Data Quality Assessment

Looking at all three datasets, what potential data quality issues do you notice? Consider missing values, data types, and any unusual patterns.

**Your Answer:**

# The 'ratings_df' data is confusing on first glance. Instead of using a CUstomerID as numbers like 1,2,3... it should be an alpha numeric ID like out UTSA abc123. It will be easier to have more once you have eclipsed a high amount of customers. The 'comments_df' data has a 'FeedbackID' and a 'CustomerID'. Having either one of those should suffice especially wince the customers contact info is also shown. In the 'sales_df' data the 'Amount' and 'Quantity' data can be confused and should be simplified or one of them should be differentiated more from the other. 

### Question 3: Analysis Readiness

Which of the three datasets appears most ready for analysis, and which would require the most preprocessing? Justify your answer.

**Your Answer:**

# The 'comments_df' data is the one most ready fro analysis. It could use some streamlining to make it better but not as much as the other two datasets. The 'sales_df' data is the second best because the data is organized well. The main thing to improve on that data would be a differentiation of the 'Amount' and 'Quantity' data. The dataset with the most preprocessing would be the 'ratings_cf'. All the data is numeric and isn't easy to understand from the first glance. THe 'CustomerID' and "RecommendationScore' data should both be changed to something simpler to understand. THe 'CustomerID' needs to be something more unique and the 'RecommendationScore' needs to be changed from a a numeric value to something else that will still be as clear. A star system or things such as "Very likely" "Unlikely" could be used for 'RecommendationScore'

---

## Submission Checklist

Before submitting, make sure you have:

- [ ] Completed all code cells in Parts 1 and 2
- [ ] Run all cells to show output
- [ ] Added your observations for each dataset
- [ ] Answered all reflection questions in Part 3
- [ ] Added your name and date at the top
- [ ] Saved the notebook
- [ ] Committed and pushed your changes to GitHub

**Great work completing your first R data analysis assignment! 🎉**

---

## 🚀 Ready to Submit?

### Easy Submission Steps (No Command Line Required!):

1. **Save this notebook** (Ctrl+S or File → Save)

2. **Use VS Code Source Control**:
   - Click the **Source Control** icon in the left sidebar (tree branch symbol)
   - Click the **"+"** button next to your notebook file
   - Type a message: `Submit homework 1 - [Your Name]`
   - Click **"Commit"** 
   - Click **"Sync Changes"** or **"Push"**

3. **Verify on GitHub**: Go to your repository online and confirm your notebook appears with your completed work

**📖 Need help?** See [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for detailed instructions.

**🎉 Congratulations on completing your first R data analysis assignment!**